### ISSUE TITLE
GraphSage Algorithm Aggregation and Weight order error

### ISSUE THREAD

**Issue by aj111000** [stellargraph/stellargraph#1789]
### Describe the bug

This isn't specifically a bug, but rather an implementation error. The order that the GraphSage Algorithm is performing the Aggregation and Weight multiplication operations is incorrect.

In the initial paper, the (pre-activation) output for vertex v at layer k is the weight matrix W_k multiplied by the output for vertex v from layer k-1 concatenated with the aggregated feature vector. 

However, stellargraph actually does this in the opposite order, first applying the weights to both the previous output vector and aggregated feature vector and then concatenating the outputs. 

This is a pretty substantial problem, because it means that the first half of the output for each layer is only based on the feature vector and the second half of the output is only based on the aggregated vector

For example, let's say that the feature vector x and aggregated vector y are both 5x1 and we want our output to be 2x1. In the standard graphsage algorithm, our weight matrix would be of the form W_k=[A B; C D] where A, B, C, and D are all 1x5 matrices, and the output would be W_k*[x;y]=[A*x+B*y; C*x+D*y], so both x and y impact both entries in the output. However, in the current implementation, we instead have two 5x1 weights X and Y and our output is [X*x; Y*y], meaning that y has no influence over the first half of the output and x has no influence over the second half of the output. 

(Note: I know that the implementation is actually a row vector multiplied by a matrix and not a matrix multiplied by a column vector, I was just using it for clarification)

I don't want this to seem like I'm being critical, you guys have made an incredible piece of software, and I only noticed this because I've been having to do a lot of work with some of the lower level functions.

### To Reproduce

Just look at the GraphSage Aggregator class

### Observed behavior

See description

### Expected behavior

See description

### Environment

N/A

### Additional context

N/A

**Comment by aj111000** [comment]
There's also a relatively easy fix that I can see. If you redefined the weights so that each weight is nxm (n for the size of the input and m for the size of the output), and then used an add operation instead of a concatenate operation, then that would solve the problem. That would be the equivalent of writing [A B; C D][x; y] = [A; C]x + [B; D]y, which is both equivalent to the original paper and just as modular as the current implementation.

**Comment by huonw** [comment]
Thanks for filing an issue. I'm not in a position to dig into the details right now, but we really appreciate that you've done so. If you feel like it, we'd be more than happy to merge a pull request with a fix, otherwise, I will try to look at this in future, but i can't guarantee when!

**Comment by aj111000** [comment]
Yeah, I can do that. I just wasn't sure how much it might mess with other people's code, since even though it's a relatively simple thing to fix it is a pretty substantial change. I'm also not sure how well it will integrate with neo4j, since I'm not very familiar with how that whole part works. I can make the changes and submit a pull request now though, and you guys can just review it

**Comment by aj111000** [comment]
I realized that it's failing several tests, but all of them are related to the intended behavior of the changes. For example, it's failing assert agg.weight_dims == [3, 2] in test_graphsage, because the goal is that the weight dims would both be 5. Is it okay if I change the testing files as well to match how the behavior should be, or is there another way to go about this?
